devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/dependency/ApplicationDeploymentClasspathBuilder.java

```diff
@@ -0,0 +1,200 @@
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ConfigurationContainer;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.ResolvedDependency;
+import org.gradle.api.artifacts.dsl.DependencyHandler;
+import org.gradle.api.plugins.JavaPlugin;
+
+import io.quarkus.gradle.tooling.ToolingUtils;
+import io.quarkus.gradle.tooling.dependency.DependencyUtils;
+import io.quarkus.gradle.tooling.dependency.ExtensionDependency;
+import io.quarkus.gradle.tooling.dependency.LocalExtensionDependency;
+import io.quarkus.maven.dependency.Dependency;
+import io.quarkus.runtime.LaunchMode;
+
+public class ApplicationDeploymentClasspathBuilder {
+
+    private static String getRuntimeConfigName(LaunchMode mode, boolean base) {
+        final StringBuilder sb = new StringBuilder();
+        sb.append("quarkus");
+        if (mode == LaunchMode.DEVELOPMENT) {
+            sb.append("Dev");
+        } else if (mode == LaunchMode.TEST) {
+            sb.append("Test");
+        } else {
+            sb.append("Prod");
+        }
+        if (base) {
+            sb.append("Base");
+        }
+        sb.append("RuntimeClasspathConfiguration");
+        return sb.toString();
+    }
+
+    public static String getBaseRuntimeConfigName(LaunchMode mode) {
+        return getRuntimeConfigName(mode, true);
+    }
+
+    public static String getFinalRuntimeConfigName(LaunchMode mode) {
+        return getRuntimeConfigName(mode, false);
+    }
+
+    public static void initConfigurations(Project project) {
+        final ConfigurationContainer configContainer = project.getConfigurations();
+
+        // Custom configuration for dev mode
+        configContainer.create(ToolingUtils.DEV_MODE_CONFIGURATION_NAME)
+                .extendsFrom(configContainer.getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME));
+
+        // Base runtime configurations for every launch mode
+        configContainer.create(ApplicationDeploymentClasspathBuilder.getBaseRuntimeConfigName(LaunchMode.TEST))
+                .extendsFrom(configContainer.getByName(JavaPlugin.TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME));
+
+        configContainer.create(ApplicationDeploymentClasspathBuilder.getBaseRuntimeConfigName(LaunchMode.NORMAL))
+                .extendsFrom(configContainer.getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME));
+
+        configContainer.create(ApplicationDeploymentClasspathBuilder.getBaseRuntimeConfigName(LaunchMode.DEVELOPMENT))
+                .extendsFrom(
+                        configContainer.getByName(ToolingUtils.DEV_MODE_CONFIGURATION_NAME),
+                        configContainer.getByName(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME),
+                        configContainer.getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME));
```

Comment:
shouldn't we also extend from `TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME ` for continuous testing?

Reply:
No, it's a different launch mode. We are resolving a separate `ApplicationModel` for each one.

Reply:
But when running `gradle quarkusDev` we only resolve one model right?

Reply:
No, we resolve both. https://github.com/quarkusio/quarkus/blob/main/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java#L331